src/directwrite/mod.rs

```diff
@@ -172,11 +166,11 @@ impl crate::Rasterize for DirectWriteRasterizer {
         let line_height = f64::from(ascent - descent + line_gap);
 
         // Since all monospace characters have the same width, we use `!` for horizontal metrics.
-        let c = '!';
-        let glyph_index = self.get_glyph_index(face, c)?;
+        let character = '!';
+        let glyph_index = self.get_glyph_index(face, character).unwrap_or(0);
```

Comment:
DirectWrite returns `0` if the glyph cannot be found in the font, but does that mean that `0` is a safe glyph index to use ourselves?

If so, does it still return the correct metrics and could we just always use that?

Reply:
It should be safe, since it's more about font design, thus the first index in a font should be 0 across the platforms. Missing glyph is also a glyph, so it's fine to use it.

Reply:
> thus the first index in a font should be 0 across the platforms

That's completely implementation dependent as far as I'm aware. And this code is exclusively about directwrite anyways.

Reply:
@davidhewitt  Do you happen to know how this `0` index story works on Windows? I can't really find anything related in docs wrt that, I just know that it 0 mean undefined according to https://docs.microsoft.com/en-us/windows/win32/api/dwrite/nf-dwrite-idwritefontface-getglyphindices . I think it should be safe to use 0 tbqh, though I can't really test it.

Reply:
Looks like there's some info here: https://docs.microsoft.com/en-us/typography/opentype/spec/recom#glyph-0-the-notdef-glyph

I can give this branch a test-drive on Windows if you tell me what I should look out for?

Reply:
I think the goal would be to see if missing glyph characters are displayed when a glyph isn't present and that normal glyphs aren't all suddenly broken.

More specific to this discussion: The question is if the notdef glyph has the same metrics as every other glyph of the font, so we could use it for the metrics instead of having to find the index for the `!` glyph.